runtime/commonMain/src/kotlinx/serialization/json/JsonOutput.kt

```diff
@@ -47,7 +49,32 @@ public interface JsonOutput: Encoder, CompositeEncoder {
     public val json: Json
 
     /**
-     * Appends given [element] to the output.
+     * Appends the given JSON [element] to the current output.
+     * This method is allowed to invoke only as the part of the whole serialization process of the class,
+     * calling this method after invoking [beginStructure] or any `encode*` method will lead to unspecified behaviour
+     * and may produce an invalid JSON result.
+     * For example:
+     * ```
+     * class Holder(val value: Int, val list: List<Int>())
+     *
+     * // Holder serialize method
+     * fun serialize(encoder: Encoder, value: Holder) {
+     *     // Completely okay, the whole Holder object is read
+     *     val jsonObject = JsonObject(...) // build a JsonObject from Holder
+     *     encoder.asJsonInput().encodeJson(jsonObject) // Write it
```

Comment:
`asJsonOutput`

Btw, these functions are internal, so maybe we shouldn't give an example which does not compile?

Reply:
Well, they matched accidentally. For the sake of simplicity,  I just used it as a pseudo-code example. Fixed it to avoid confusion